tests/integration/security/mtls_cert_securenaming/mtls_cert_securenaming_test.go

```diff
@@ -0,0 +1,244 @@
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/resource/environment"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/tests/integration/security/util"
+	"istio.io/istio/tests/integration/security/util/cert"
+	"istio.io/istio/tests/integration/security/util/connection"
+)
+
+const (
+	// The length of the example certificate chain.
+	exampleCertChainLength = 3
+
+	defaultIdentityDR = `apiVersion: "networking.istio.io/v1alpha3"
+kind: "DestinationRule"
+metadata:
+  name: "service-b-dr"
+spec:
+  host: "b.NS.svc.cluster.local"
+  trafficPolicy:
+    tls:
+      mode: ISTIO_MUTUAL
+      subjectAltNames:
+      - "spiffe://cluster.local/ns/NS/sa/default"
+`
+	correctIdentityDR = `apiVersion: "networking.istio.io/v1alpha3"
+kind: "DestinationRule"
+metadata:
+  name: "service-b-dr"
+spec:
+  host: "b.NS.svc.cluster.local"
+  trafficPolicy:
+    tls:
+      mode: ISTIO_MUTUAL
+      subjectAltNames:
+      - "spiffe://cluster.local/ns/NS/sa/b"
+`
+	nonExistIdentityDR = `apiVersion: "networking.istio.io/v1alpha3"
+kind: "DestinationRule"
+metadata:
+  name: "service-b-dr"
+spec:
+  host: "b.NS.svc.cluster.local"
+  trafficPolicy:
+    tls:
+      mode: ISTIO_MUTUAL
+      subjectAltNames:
+      - "I-do-not-exist"
+`
+	identityListDR = `apiVersion: "networking.istio.io/v1alpha3"
+kind: "DestinationRule"
+metadata:
+  name: "service-b-dr"
+spec:
+  host: "b.NS.svc.cluster.local"
+  trafficPolicy:
+    tls:
+      mode: ISTIO_MUTUAL
+      subjectAltNames:
+      - "spiffe://cluster.local/ns/NS/sa/a"
+      - "spiffe://cluster.local/ns/NS/sa/b"
+      - "spiffe://cluster.local/ns/default/sa/default"
+      - "I-do-not-exist"
+`
+)
+
+// TestMTLSCertSecureNaming verifies:
+// - The certificate issued by CA to the sidecar is as expected and that strict mTLS works as expected.
+// - The CA certificate in the configmap of each namespace is as expected, which
+//   is used for data plane to control plane TLS authentication.
+func TestMTLSCertSecureNaming(t *testing.T) {
+	framework.NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			istioCfg := istio.DefaultConfigOrFail(t, ctx)
+			testNamespace := namespace.NewOrFail(t, ctx, namespace.Config{
+				Prefix: "test-mtls",
+				Inject: true,
+			})
+			namespace.ClaimOrFail(t, ctx, istioCfg.SystemNamespace)
+			// Check that the CA certificate in the configmap of each namespace is as expected, which
+			// is used for data plane to control plane TLS authentication.
+			retry.UntilSuccessOrFail(t, func() error {
+				return checkCACert(ctx, t, testNamespace)
+			}, retry.Delay(time.Second), retry.Timeout(10*time.Second))
+			var a, b echo.Instance
+			echoboot.NewBuilderOrFail(ctx, ctx).
+				With(&a, util.EchoConfig("a", testNamespace, false, nil, g, p)).
+				With(&b, util.EchoConfig("b", testNamespace, false, nil, g, p)).
+				BuildOrFail(t)
+
+			ctx.NewSubTest("mTLS cert validation").
```

Comment:
How about we extract this plugin CA cert validation into a separate TestPluginCACert() method?

Reply:
I changed the name. This test is mostly validating the cert chain length and the root cert matching. So it's pretty much validating cert presents and the plugin CA root is used. I don't think we need to further splitting the test, unless we make the tests validating more things. I think this is good enough for now.